From 696c4498dd628c2db9ef2db0f2f871c34ea8f9b6 Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Thu, 27 Nov 2014 11:46:40 -0500
Subject: [fs] ovl: rename filesystem type to "overlay"

Message-id: <20141127114640.9170.62069.stgit@warthog.procyon.org.uk>
Patchwork-id: 100984
O-Subject: [RHEL7 PATCH 1/7] ovl: rename filesystem type to "overlay"
Bugzilla: 985875
RH-Acked-by: Alexander Viro <aviro@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

From: Miklos Szeredi <mszeredi@suse.cz>

Some distributions carry an "old" format of overlayfs while mainline has a
"new" format.

The distros will possibly want to keep the old overlayfs alongside the new
for compatibility reasons.

To make it possible to differentiate the two versions change the name of
the new one from "overlayfs" to "overlay".

Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
Reported-by: Serge Hallyn <serge.hallyn@ubuntu.com>
Cc: Andy Whitcroft <apw@canonical.com>
Original-commit: ef94b1864d1ed5be54376404bb23d22ed0481feb
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/filesystems/overlayfs.txt b/Documentation/filesystems/overlayfs.txt
index 530850a..a27c950 100644
--- a/Documentation/filesystems/overlayfs.txt
+++ b/Documentation/filesystems/overlayfs.txt
@@ -64,7 +64,7 @@ is formed.
 At mount time, the two directories given as mount options "lowerdir" and
 "upperdir" are combined into a merged directory:
 
-  mount -t overlayfs overlayfs -olowerdir=/lower,upperdir=/upper,\
+  mount -t overlay overlay -olowerdir=/lower,upperdir=/upper,\
 workdir=/work /merged
 
 The "workdir" needs to be an empty directory on the same filesystem
diff --git a/MAINTAINERS b/MAINTAINERS
index 693910f..4e6a367 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -6091,7 +6091,7 @@ F: drivers/scsi/osd/
 F: include/scsi/osd_*
 F: fs/exofs/
 
-OVERLAYFS FILESYSTEM
+OVERLAY FILESYSTEM
 M: Miklos Szeredi <miklos@szeredi.hu>
 L: linux-fsdevel@vger.kernel.org
 S: Supported
diff --git a/fs/Makefile b/fs/Makefile
index 9506dd7..2d0e6bd 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -105,7 +105,7 @@ obj-$(CONFIG_QNX6FS_FS)  += qnx6/
 obj-$(CONFIG_AUTOFS4_FS) += autofs4/
 obj-$(CONFIG_ADFS_FS)  += adfs/
 obj-$(CONFIG_FUSE_FS)  += fuse/
-obj-$(CONFIG_OVERLAYFS_FS) += overlayfs/
+obj-$(CONFIG_OVERLAY_FS) += overlayfs/
 obj-$(CONFIG_UDF_FS)  += udf/
 obj-$(CONFIG_SUN_OPENPROMFS) += openpromfs/
 obj-$(CONFIG_OMFS_FS)  += omfs/
diff --git a/fs/overlayfs/Kconfig b/fs/overlayfs/Kconfig
index e601259..3435581 100644
--- a/fs/overlayfs/Kconfig
+++ b/fs/overlayfs/Kconfig
@@ -1,4 +1,4 @@
-config OVERLAYFS_FS
+config OVERLAY_FS
  tristate "Overlay filesystem support"
  help
    An overlay filesystem combines two filesystems - an 'upper' filesystem
diff --git a/fs/overlayfs/Makefile b/fs/overlayfs/Makefile
index 8f91889..900daed 100644
--- a/fs/overlayfs/Makefile
+++ b/fs/overlayfs/Makefile
@@ -2,6 +2,6 @@
 # Makefile for the overlay filesystem.
 #
 
-obj-$(CONFIG_OVERLAYFS_FS) += overlayfs.o
+obj-$(CONFIG_OVERLAY_FS) += overlay.o
 
-overlayfs-objs := super.o inode.o dir.o readdir.o copy_up.o
+overlay-objs := super.o inode.o dir.o readdir.o copy_up.o
diff --git a/fs/overlayfs/super.c b/fs/overlayfs/super.c
index 1620eb3..aad1e9e 100644
--- a/fs/overlayfs/super.c
+++ b/fs/overlayfs/super.c
@@ -24,7 +24,7 @@ MODULE_AUTHOR("Miklos Szeredi <miklos@szeredi.hu>");
 MODULE_DESCRIPTION("Overlay filesystem");
 MODULE_LICENSE("GPL");
 
-#define OVERLAYFS_SUPER_MAGIC 0x794c764f
+#define OVERLAYFS_SUPER_MAGIC 0x794c7630
 
 struct ovl_config {
  char *lowerdir;
@@ -786,11 +786,11 @@ static struct dentry *ovl_mount(struct file_system_type *fs_type, int flags,
 
 static struct file_system_type ovl_fs_type = {
  .owner  = THIS_MODULE,
- .name  = "overlayfs",
+ .name  = "overlay",
  .mount  = ovl_mount,
  .kill_sb = kill_anon_super,
 };
-MODULE_ALIAS_FS("overlayfs");
+MODULE_ALIAS_FS("overlay");
 
 static int __init ovl_init(void)
 {
-- 
1.7.1